rotkehlchen/db/history_events.py

```diff
@@ -174,13 +263,30 @@ def edit_history_event(self, write_cursor: 'DBCursor', event: HistoryBaseEntry)
             else:  # all other data
                 write_cursor.execute(f'{updatestr} WHERE identifier=?', (*bindings, event.identifier))  # noqa: E501
 
-        # Also mark it as customized
         write_cursor.execute(
             'INSERT OR IGNORE INTO history_events_mappings(parent_identifier, name, value) '
             'VALUES(?, ?, ?)',
             (event.identifier, HISTORY_MAPPING_KEY_STATE, HISTORY_MAPPING_STATE_CUSTOMIZED),
         )
 
+        # Track modification only if balance-affecting fields changed. cannot be None here
+        if old_data == (
+            event.timestamp,
+            event.asset.identifier,
+            str(event.amount),
+            event.event_type.serialize(),
+            event.event_subtype.serialize(),
+            event.location_label,
+        ):
+            return
+
+        old_ts, old_asset, _, _, _, _ = old_data
```

Comment:
there is one scenario that I believe is not possible if everything goes okey since you have `if write_cursor.rowcount != 1:` but if for whatever reason `old_data` is None then this raises an exception. In a multi thread scenario

1. thread 1 calls the edit and old_data is None
2. thread 2 creates the event
3. thread 1 does the edit and reaches here. Fails